### PR TITLE
Fix Apache Arrow support.

### PR DESCRIPTION
**Pull request by KalyanovD**
# Description

Changes proposed in this pull request:
- Fix for supporting newer version of Apache Arrow 
- Apache Arrow sample has been updated. Support for Win and Mac has been added.

### CODE REVIEW

**Comment by KalyanovD** [comment]
make test

**Comment by KalyanovD** [comment]
restart

File: samples/cpp/arrow/makefile_lnx [hunk 1]
```diff
@@ -50,19 +50,19 @@ ifndef compiler
...
 ifndef DAALROOT
-    DAALROOT = ./../..
+    DAALROOT = ./../../..
 endif
 
 ifndef TBBROOT
     TBBROOT = ./../../../../tbb/latest
 endif
-TBB_PATH = "$(TBBROOT)/lib/$(_IA)/gcc4.8"
+TBB_PATH = $(TBBROOT)/lib/$(_IA)/gcc4.8
```
Comment (averbukh) [inline samples/cpp/arrow/makefile_lnx]: Revert quotes (at least to be consistent with other samples).

File: samples/cpp/arrow/makefile_lnx [hunk 2]
```diff
@@ -50,19 +50,19 @@ ifndef compiler
...
 
 ifndef sample
     sample = $(ARROW)
 endif
 
-DAAL_PATH = "$(DAALROOT)/lib/$(_IA)"
+DAAL_PATH = $(DAALROOT)/lib/$(_IA)
```
Comment (averbukh) [inline samples/cpp/arrow/makefile_lnx]: Revert quotes (at least to be consistent with other samples).

File: samples/cpp/arrow/readme.html [hunk 3]
```diff
@@ -930,34 +930,73 @@ <h1 class="title">Intel® Data Analytics Acceleration Library Apache Arrow* Samp
...
   <p>Unzip the archive with Intel® DAAL samples to your working directory (<span class="filepath"><var>&lt;sample_dir&gt;</var></span>).
 <h2><a name="System_Requirements">System Requirements</a></h2>
-<p>You can use Intel® DAAL Apache Arrow samples on Linux* operating system.
+<p>You can use Intel® DAAL Apache Arrow* samples on Linux*, Windows* and MacOS* operating systems.
 For a list of Intel® DAAL hardware and software requirements, refer to release notes
 for the version of Intel® DAAL you are using. </p>
     <h3>Apache Arrow* implementations against which Intel® DAAL has been validated:</h3>
     <ul>
-        <li>Apache Arrow* 0.13.0</li>
+        <li>Apache Arrow* 0.15.0</li>
```
Comment (averbukh) [inline samples/cpp/arrow/readme.html]: We have 0.15.1 already.

File: samples/cpp/arrow/readme.html [hunk 4]
```diff
@@ -930,34 +930,73 @@ <h1 class="title">Intel® Data Analytics Acceleration Library Apache Arrow* Samp
...
     </ul>
-    <p><strong>Note:</strong> Intel® DAAL is expected to work on other Apache Arrow* version
+    <p><strong>Note:</strong> Intel® DAAL is expected to work on higher Apache Arrow* versions
```
Comment (averbukh) [inline samples/cpp/arrow/readme.html]: Why your expected DAAL to work with HIGHER versions???

File: samples/cpp/arrow/readme.html [hunk 5]
```diff
@@ -930,34 +930,73 @@ <h1 class="title">Intel® Data Analytics Acceleration Library Apache Arrow* Samp
...
 <h3> Setting Up the Build Environment&nbsp;</h3>
 <p> Before you build the sample, you must set certain environment variables that define the location of related libraries. Intel® DAAL includes the <span class="filepath">vars</span> scripts that you can run to set environment variables:</p>
     <ul id="GUID-2FEE5E18-4507-402C-806D-F3C422B0D085">
+        <li>On Windows*, you can find the <span class="filepath">vars.bat</span> batch file at <span class="filepath">&lt;install-dir&gt;\compilers_and_libraries_xxxx.x.xxx\windows\daal\bin\:</span><samp class="codeph"><br />
+            vars.bat intel64
+            </samp>
+        </li>
         <li>On Linux*, you can find the <span class="filepath">vars.sh</span> shell script at <span class="filepath">&lt;install-dir&gt;\compilers_and_libraries_xxxx.x.xxx\linux\daal\bin:</span><samp class="codeph"><br />
-            source vars.sh {ia32|intel64}
+            source vars.sh intel64
```
Comment (averbukh) [inline samples/cpp/arrow/readme.html]: Why to specify intel64, if it's the only option?

**Comment by KalyanovD** [comment]
make test

File: samples/cpp/arrow/makefile_lnx (hunk 1 above)
Comment (KalyanovD) [reply samples/cpp/arrow/makefile_lnx]: Revert was done for both comments.

File: samples/cpp/arrow/makefile_lnx (hunk 2 above)
Comment (KalyanovD) [reply samples/cpp/arrow/makefile_lnx]: Revert was done for both comments.

File: samples/cpp/arrow/readme.html (hunk 3 above)
Comment (KalyanovD) [reply samples/cpp/arrow/readme.html]: Sample was successfully built and tested with 0.15.1 version. Readme files were changed.

File: samples/cpp/arrow/readme.html (hunk 4 above)
Comment (KalyanovD) [reply samples/cpp/arrow/readme.html]: Changes were revert to the previous version.

File: samples/cpp/arrow/readme.html (hunk 5 above)
Comment (KalyanovD) [reply samples/cpp/arrow/readme.html]: Comment concerning support of x64 architectures only was added below.

File: samples/cpp/arrow/README.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+Unzip the archive with Intel® DAAL samples to your working directory (`<sample_dir>`).
+
+## System Requirements
+You can use Intel(R) DAAL Apache Arrow\* samples on Linux\*, Windows\*, and macOS\* operating systems. For a list of Intel(R) DAAL hardware and software requirements, refer to release notes for the version of Intel(R) DAAL you are using.
+
+### Apache Arrow\* implementations against which Intel(R) DAAL has been validated:
+- Apache Arrow\* 0.15.1
+
+**Note:** Intel(R) DAAL is expected to work on other Apache Arrow\* version as well. Let us know if you have any troubles with the distribution you are using.
```
Comment (outoftardis) [inline samples/cpp/arrow/README.md]: versions

File: include/data_management/data/arrow_numeric_table.h [hunk 7]
```diff
@@ -163,9 +164,13 @@ class DAAL_EXPORT ArrowImmutableNumericTable : public BaseArrowImmutableNumericT
         {
             const NumericTableFeature & f = (*_ddict)[i];
 
+#if ARROW_VERSION >= 15000
```
Comment (averbukh) [inline include/data_management/data/arrow_numeric_table.h]: I see some copy-paste here. Could you please introduce function to encapsulate differences between Arrow versions?

File: samples/cpp/arrow/license.txt [hunk 8]
```diff
@@ -0,0 +1,10 @@
+Apache Arrow* Samples
+
+License Information Follows
+---------------------------
+* *.* all of source codes and data file 
+* readme.html, makefile_*, *.bat,*.lst
+
+Copyright 2015-2019 Intel Corporation
```
Comment (averbukh) [inline samples/cpp/arrow/license.txt]: -2020 ?

File: samples/cpp/arrow/makefile_mac [hunk 9]
```diff
@@ -0,0 +1,115 @@
+#===============================================================================
+# Copyright 2017-2019 Intel Corporation
```
Comment (averbukh) [inline samples/cpp/arrow/makefile_mac]: -2020 ?

File: include/data_management/data/arrow_numeric_table.h (hunk 7 above)
Comment (KalyanovD) [reply include/data_management/data/arrow_numeric_table.h]: Such a function was introduced. Thank you for comment!

File: samples/cpp/arrow/license.txt (hunk 8 above)
Comment (KalyanovD) [reply samples/cpp/arrow/license.txt]: Copyrights were changed

File: samples/cpp/arrow/makefile_mac (hunk 9 above)
Comment (KalyanovD) [reply samples/cpp/arrow/makefile_mac]: Copyrights were changed

**Pull request merged**